run.rb

```diff
@@ -1,26 +1,32 @@
-require 'rubygems'
-require 'bundler/setup'
-require 'dropbox_sdk'
+require 'dropbox'
 require 'dotenv'
 Dotenv.load
 
-dropbox = DropboxClient.new(ENV['API_KEY'])
-uploads = dropbox.metadata(ENV['UPLOADS'], 25_000, true)['contents']
+dropbox = Dropbox::Client.new(ENV['API_KEY'])
+uploads = dropbox.list_folder(ENV['UPLOADS']).entries
 
 uploads.each do |upload|
-  unless upload['is_dir'] 
-    folder = upload['path'].match(/[0-9]{4}\-[0-9]{2}/).to_s
-    new_folder = "/#{ENV['UPLOADS']}/#{folder}"
+  unless upload.is_a?(Dropbox::FolderMetadata)
+    puts "Considering #{upload.path_display}"
+
+    folder = upload.path_display.match(/[0-9]{4}\-[0-9]{2}/).to_s
+    if folder == ''
+      puts "Cannot extract target folder name from #{upload.path_display} so skipping"
+      next
+    end
+
+    new_folder = "#{ENV['UPLOADS']}/#{folder}"
 
     begin
```

Comment:
@maxmanders It also looks like with the new API you don't have to create the folder first, I removed this whole block locally and the move endpoint will just create the new folder since you're specifying a file within it so it knows it's a folder.